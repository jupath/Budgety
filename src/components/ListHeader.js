import React, { Component } from 'react';

class ListHeader extends Component {
  state = {

  };

  render() {
    return (
      <div>
      ListHeader
      </div>
    );
  }
}

export default ListHeader;
