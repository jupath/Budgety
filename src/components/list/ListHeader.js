import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const ListHeader = props => (
  <div className="list__header py-3 px-2">
    <div>
      <h4 className="mb-0">{props.listName}</h4>
    </div>
    <div>
      <Link className="btn btn-info" to={`/add/${props.itemsType}`}>Add new {props.itemsType}</Link>
    </div>
  </div>
);

ListHeader.propTypes = {
  itemsType: PropTypes.string.isRequired,
  listName: PropTypes.string.isRequired,
};

export default ListHeader;
