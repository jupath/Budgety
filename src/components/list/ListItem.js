import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const ListItem = props => (
  <Link to={`/edit/${props.listItem.id}`}>
    <div className="list__item">
      <div>
        {props.listItem.description}<br />
        {moment(props.listItem.date).format('MMMM Do, YYYY')}
      </div>
      <div>
        {props.listItem.amount}
      </div>
    </div>
  </Link>
);

ListItem.propTypes = {
  listItem: PropTypes.shape({
    itemType: PropTypes.string,
    description: PropTypes.string,
    amount: PropTypes.number,
    date: PropTypes.number,
    note: PropTypes.string,
    id: PropTypes.string,
  }).isRequired,
};

export default ListItem;
