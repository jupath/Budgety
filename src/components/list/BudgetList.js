import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem';

export const BudgetList = props => (
  <div>
    {
      props.items.map(item => <ListItem key={item.id} listItem={item} />)
    }
  </div>
);

BudgetList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default BudgetList;
