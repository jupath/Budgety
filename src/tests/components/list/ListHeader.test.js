import React from 'react';
import { shallow } from 'enzyme';
import { ListHeader } from '../../../components/list/ListHeader';

test('should render BudgetList correctly', () => {
  const wrapper = shallow(<ListHeader itemsType='expense' listName='Expenses' />);
  expect(wrapper).toMatchSnapshot();
});
