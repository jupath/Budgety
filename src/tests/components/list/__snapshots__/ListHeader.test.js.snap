// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BudgetList correctly 1`] = `
<div
  className="list__header py-3 px-2"
>
  <div>
    <h4
      className="mb-0"
    >
      Expenses
    </h4>
  </div>
  <div>
    <Link
      className="btn btn-info"
      replace={false}
      to="/add/expense"
    >
      Add new 
      expense
    </Link>
  </div>
</div>
`;
